From 0dd50c4c0a30786462f64a4239a4e97a0e437deb Mon Sep 17 00:00:00 2001
From: "qiang.cheng" <qiang.cheng@amlogic.com>
Date: Tue, 4 Jul 2023 14:39:31 +0800
Subject: [PATCH 05/10] yocto: add build config for s4_s905y4 kernel 5.15 [1/1]

PD#SWPL-130509

Problem:
bringup s4_s905y4 kernel 5.15 on smarthome

Solution:
add yocto build config

Verify:
local

Signed-off-by: qiang.cheng <qiang.cheng@amlogic.com>
Change-Id: Idadd6c33f9f9976fafc7079f1758c89ad0f30519
---
 conf/machine/include/mesons4_k5.15_64b.inc    | 13 ++++
 conf/machine/mesons4-ap222-5.15.conf          |  1 +
 conf/machine/mesons4-ap222-weston-5.15.config | 73 +++++++++++++++++++
 recipes-kernel/linux/5.15/s4.cfg              | 15 ++++
 4 files changed, 102 insertions(+)
 create mode 100644 conf/machine/include/mesons4_k5.15_64b.inc
 create mode 120000 conf/machine/mesons4-ap222-5.15.conf
 create mode 100644 conf/machine/mesons4-ap222-weston-5.15.config
 create mode 100644 recipes-kernel/linux/5.15/s4.cfg

diff --git a/conf/machine/include/mesons4_k5.15_64b.inc b/conf/machine/include/mesons4_k5.15_64b.inc
new file mode 100644
index 00000000..8211ca47
--- /dev/null
+++ b/conf/machine/include/mesons4_k5.15_64b.inc
@@ -0,0 +1,13 @@
+# common configuration
+require conf/machine/include/amlogic_common.inc
+require conf/machine/include/amlogic_64b.inc
+
+PREFERRED_VERSION_u-boot = "v2019.%"
+PREFERRED_VERSION_linux-meson = "5.15.%"
+PREFERRED_VERSION_media-modules = "5.15%"
+PREFERRED_VERSION_qca6174 = "5.15%"
+PREFERRED_PROVIDER_virtual/gpu ?= "dvalin"
+OVERRIDES .= ":k5.15"
+LINUXLIBCVERSION ?= "5.15%"
+HOSTTOOLS_NONFATAL += " pahole"
+OVERRIDES .= ":s4"
diff --git a/conf/machine/mesons4-ap222-5.15.conf b/conf/machine/mesons4-ap222-5.15.conf
new file mode 120000
index 00000000..6360e6d6
--- /dev/null
+++ b/conf/machine/mesons4-ap222-5.15.conf
@@ -0,0 +1 @@
+mesons4-ap222-weston-5.15.config
\ No newline at end of file
diff --git a/conf/machine/mesons4-ap222-weston-5.15.config b/conf/machine/mesons4-ap222-weston-5.15.config
new file mode 100644
index 00000000..9e75e819
--- /dev/null
+++ b/conf/machine/mesons4-ap222-weston-5.15.config
@@ -0,0 +1,73 @@
+#@TYPE: Machine
+#@NAME: meson
+
+#@DESCRIPTION: Machine configuration for meson systems
+
+include conf/machine/include/mesons4_k5.15_64b.inc
+include conf/machine/include/amlogic_weston.inc
+
+PREFERRED_PROVIDER_virtual/mesa = "libgles-eabihf-dvalin-wayland-drm"
+PREFERRED_PROVIDER_virtual/mesa-gl = "libgles-eabihf-dvalin-wayland-drm"
+PREFERRED_PROVIDER_virtual/libgbm = "libgles-eabihf-dvalin-wayland-drm"
+PREFERRED_PROVIDER_virtual/egl = "libgles-eabihf-dvalin-wayland-drm"
+PREFERRED_PROVIDER_virtual/libgl = "libgles-eabihf-dvalin-wayland-drm"
+PREFERRED_PROVIDER_virtual/libgles1 = "libgles-eabihf-dvalin-wayland-drm"
+PREFERRED_PROVIDER_virtual/libgles2 = "libgles-eabihf-dvalin-wayland-drm"
+
+MACHINE_ESSENTIAL_EXTRA_RRECOMMENDS += " \
+        gpu \
+        wififw-qca6174 \
+        qca6174 \
+        aucpu-fw \
+        coreutils \
+        "
+
+RDEPENDS_packagegroup-amlogic-baserootfs_remove = " aml-subtitleserver"
+RDEPENDS_packagegroup-amlogic-baserootfs_remove = " gst-aml-drm-plugins"
+RDEPENDS_packagegroup-amlogic-baserootfs_remove = " fuse-exfat"
+
+CHIPSET_NAME = "S905Y4"
+KERNEL_DEVICETREE = "s4_s905y4_ap222_linux.dtb"
+UBOOT_MACHINE = "s4_ap222_config"
+KERNEL_MODULE_AUTOLOAD += " btqca"
+KERNEL_MODULE_PROBECONF += "btqca"
+KERNEL_MODULE_AUTOLOAD += " snd-soc-dummy_codec"
+KERNEL_MODULE_AUTOLOAD += " snd-soc-aml_t9015"
+KERNEL_MODULE_AUTOLOAD += " meson_ir"
+KERNEL_MODULE_AUTOLOAD += " hci_uart"
+KERNEL_MODULE_PROBECONF += "hci_uart"
+KERNEL_MODULE_AUTOLOAD += "aml_aucpu"
+KERNEL_MODULE_AUTOLOAD += " media_sync"
+KERNEL_MODULE_AUTOLOAD += "meson-cpufreq"
+
+DISTRO_FEATURES_append = " drm"
+DISTRO_FEATURES_append = " gpt-partition"
+DISTRO_FEATURES_append = " kernel_515"
+
+#DISTRO_FEATURES_append = " onepass"
+DISTRO_FEATURES_append = " secure-u-boot optee "
+#DISTRO_FEATURES_append = " widevine "
+#DISTRO_FEATURES_append = " amlogic-tv "
+DISTRO_FEATURES_append = " amlogic-dvb "
+DISTRO_FEATURES_append = " alsa "
+DISTRO_FEATURES_append = " pam"
+DISTRO_FEATURES_BACKFILL_CONSIDERED = "sysvinit"
+DEFAULT_DISTRO_FEATURES += " systemd"
+DISTRO_FEATURES_append = " systemd"
+DISTRO_FEATURES_remove = " sysvinit"
+#DISTRO_FEATURES_append = " opengl"
+DISTRO_FEATURES_append = " gstreamer1"
+DISTRO_FEATURES_append = " aml-dtv "
+#DISTRO_FEATURES_append = " qt5"
+#DISTRO_FEATURES_append = " qt5-tests"
+#DISTRO_FEATURES_append = " playready "
+DISTRO_FEATURES_append = " softap"
+DISTRO_FEATURES_append = " aml-libgdc"
+
+OVERRIDES .= ":ap222:s4:onepass"
+MACHINEOVERRIDES .= ":client"
+
+EXTRA_BLUETOOTH_STUFF := "${@bb.utils.contains('DISTRO_FEATURES', 'bluetooth', bb.utils.contains('DISTRO_FEATURES', 'bluez5', ' bluez5', ' bluez4', d), '', d)}"
+EXTRA_BLUETOOTH_STUFF += " bluez-alsa"
+DISTRO_FEATURES_append = "${EXTRA_BLUETOOTH_STUFF}"
+
diff --git a/recipes-kernel/linux/5.15/s4.cfg b/recipes-kernel/linux/5.15/s4.cfg
new file mode 100644
index 00000000..8adf506a
--- /dev/null
+++ b/recipes-kernel/linux/5.15/s4.cfg
@@ -0,0 +1,15 @@
+CONFIG_USERFAULTFD=y
+CONFIG_AUTOFS4_FS=y
+CONFIG_VT=y
+CONFIG_CONSOLE_TRANSLATIONS=y
+CONFIG_HW_CONSOLE=y
+CONFIG_FHANDLE=y
+CONFIG_SERIAL_DEV_BUS=y
+CONFIG_SERIAL_DEV_CTRL_TTYPORT=y
+CONFIG_SYSVIPC=y
+CONFIG_BT_RFCOMM=y
+# CONFIG_CC_HAVE_STACKPROTECTOR_SYSREG is not set
+# CONFIG_BT_HCIBTSDIO is not set
+CONFIG_DRM_MESON_EMULATE_FBDEV=y
+CONFIG_EXFAT_FS=y
+
-- 
2.25.1

